### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by dnut**
Addresses #43

Added docs to explain how to get up and running with Sig.

- Added new content to readme.md:
  - Links to blog posts and other docs in the repo
  - Setting up a development environment
  - Building Sig
  - Running Sig: subcommands (e.g. gossip) and CLI options
  - Running the tests and benchmarks
- Moved API documentation into docs/api.md and linked to it from the "Import Sig" section of the readme.
- Fixed incorrect license and zig version icons in the readme header
- Fixed inconsistent sig version in build.zig.zon (to be consistent with cmd.zig)

See the updated readme in a realistic looking context: https://github.com/dnut/sig/tree/docs-improvements

Review Prompts:
1. Should we just publish the readme to github.io as is, or should we create another page to publish there?
2. Do the section structure and section headings make sense?
3. Should I move any content from the readme to CONTRIBUTING.md? For example, environment setup or running tests. I feel it is more discoverable in the readme.
4. Should I change the header of the readme, for example adding a logo?

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,61 +1,210 @@
...
 ⚠️ NOTE: This is a WIP, please open any issues for any bugs/improvements.
 
-## Why Zig?
+## 📋 Setup
+
+Zig 0.11.0 is required to build Sig.
+
+<details><summary>Dev Environment Details</summary>
+
+### Build Dependencies
```
Comment (0xNineteen) [inline readme.md:27]: should probs move this section out of the drop menu - just have it below setup instead of the sentence

File: readme.md [hunk 2]
```diff
@@ -125,2440 +274,51 @@ Add `Sig` to your Zig project using `build.zig.zon` file (available for Zig >= 0
...
-</details>
+- **[Gossip](src/gossip)** - A gossip spy node, run by: `sig gossip` or `zig build run -- gossip`
 
-<details>
-<summary><code>getBlockTime</code> - Returns the estimated production time of a block</summary>
-<br/>
+- **[Core](src/core)** - Core data structures shared across modules
 
-**Params:** <code>(slot: u64)</code>
+- **[RPC Client](src/rpc)** - A fully featured HTTP RPC client with ability to query all on-chain data along with sending transactions
```
Comment (0xNineteen) [inline readme.md:308]: probs want to also link the `docs/api.md` somewhere here too

Comment (0xNineteen) [reply readme.md:308]: probs wana also include a link/section for `Contributing`

Comment (dnut) [reply readme.md:307]: I included a section in the readme called Develop that describes running tests and benchmarks, and it links to the Setup section and CONTRIBUTING.md. Do you think there's a better way to organize this?

**Review by 0xNineteen — APPROVED** [review]

**Review by ultd — DISMISSED** [review]

**Pull request merged**